editor/src/messages/portfolio/document/node_graph/node_graph_message_handler.rs

```diff
@@ -903,6 +903,69 @@ impl<'a> MessageHandler<NodeGraphMessage, NodeGraphHandlerData<'a>> for NodeGrap
 						};
 						responses.add(FrontendMessage::UpdateWirePathInProgress { wire_path: Some(wire_path) });
 					}
+				} else if self.disconnecting.is_some() {
+					// Disconnecting with no upstream node, create new value node.
+					let to_connector = network_interface.input_connector_from_click(ipp.mouse.position, selection_network_path);
+					if let Some(to_connector) = &to_connector {
+						let Some(input_position) = network_interface.input_position(to_connector, selection_network_path) else {
+							log::error!("Could not get input position for connector: {to_connector:?}");
+							return;
+						};
+						self.wire_in_progress_to_connector = Some(input_position);
+					}
+					// Not hovering over a node input or node output, insert the node
+					else {
+						// Disconnect if the wire was previously connected to an input
+						if let Some(disconnecting) = self.disconnecting.take() {
+							let mut position = if let Some(to_connector) = self.wire_in_progress_to_connector { to_connector } else { point };
+							// Offset to drag from center of node
+							position = position - DVec2::new(24. * 3., 24.);
+
+							// Offset to account for division rounding error
+							if position.x < 0. {
+								position.x = position.x - 1.;
+							}
+							if position.y < 0. {
+								position.y = position.y - 1.;
+							}
+
+							let Some(input) = network_interface.take_input(&disconnecting, breadcrumb_network_path) else {
+								return;
+							};
+
+							let drag_start = DragStart {
+								start_x: point.x,
+								start_y: point.y,
+								round_x: 0,
+								round_y: 0,
+							};
+
+							self.drag_start = Some((drag_start, false));
+							self.node_has_moved_in_drag = false;
+							self.update_node_graph_hints(responses);
+
+							let node_id = NodeId::new();
+							responses.add(NodeGraphMessage::CreateNodeFromContextMenu {
+								node_id: Some(node_id),
+								node_type: "Identity".to_string(),
```

Comment:
Why always the identity node? Surely the relevant value node should be preferred. For example using the identity node, it is not possible to edit the value of an f64, whereas you can with a number value node.

Reply:
I agree, I made a general purpose value node to replace the identity node